extensions/wikia/EnableDiscussions/EnableDiscussionsHooks.class.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+class EnableDiscussionsHooks {
+
+	const DISCUSSIONS_URL = '/d/f';
+	// Forum namespaces should be available
+	// because both Forum.setup and ForumDisabled.setup define them
+	private static $forumNamespaces = [
+		NS_WIKIA_FORUM_BOARD,
+		NS_WIKIA_FORUM_TOPIC_BOARD,
+		NS_WIKIA_FORUM_BOARD_THREAD,
+	];
+
+
+	public static function onBeforeInitialize(
+		Title $title = null, $unused, OutputPage $output = null, $user, $request, $mediaWiki
+	) {
+		if ( !self::shouldRedirectToDiscussions() || $title == null ) {
+			return true;
+		}
+		$forumTitle = self::getForumTitle( $title );
+		if ( empty( $forumTitle ) ) {
+			return true;
+		}
+
+		$output->redirect( self::getDiscussionsUrl( $forumTitle->getNamespace(),
+			$forumTitle->getArticleID() ) );
+
+		return false;
+	}
+
+	/**
+	 * @return bool
+	 */
+	private static function shouldRedirectToDiscussions() {
+		global $wgEnableForumExt, $wgEnableDiscussions;
+
+		return empty( $wgEnableForumExt ) && $wgEnableDiscussions;
+	}
+
+	/**
+	 * @param $title
+	 * @return null|Title
+	 */
+	public static function getForumTitle( Title $title ) {
+		if ( $title->isSpecial( 'Forum' ) ||
+		     in_array( $title->getNamespace(), self::$forumNamespaces )
+		) {
+			return $title;
+		}
+		if ( $title->getNamespace() === NS_USER_WALL_MESSAGE ) {
+			$mainTitle = Title::newFromID( $title->getText() );
+			/*
+			 * if we visit the thread using Thread:xxxxx url, $wgTitle's namespace is 1201 for both, wall and forum
+			 * threads. But when we create new title from it using newFromId, we're able to distinguish those two
+			 */
+			if ( $mainTitle instanceof Title &&
+			     in_array( $mainTitle->getNamespace(), self::$forumNamespaces )
+			) {
+				return $mainTitle;
+			}
+		}
+		if ( $title->getNamespace() === 0 ) {
```

Comment:
I believe that both Wall and Forum namespaces should have been loaded since, they are included in the WallDisabled.setup.php and ForumDisabled.setup.php

Therefore the namespaces should be available, due to config:
```
if( !empty( $wgEnableWallEngine ) ) {
	include("$IP/extensions/wikia/Wall/Wall.setup.php");
} else {
	include("$IP/extensions/wikia/Wall/WallDisabled.setup.php");
}
``` 

```
if ( !empty( $wgEnableForumExt ) ) {
        $wgEnableCommentsIndex = true;
        include( "{$IP}/extensions/wikia/Forum/Forum.setup.php" );
} else {
        include( "{$IP}/extensions/wikia/Forum/ForumDisabled.setup.php" );
}
```

But as you've deducted theres an issue with paths like '/wiki/Board:Fun_And_Games' and they've got NS_MAIN namespace, at least in the hook.